src/envoy/server/alembic/versions/33a17317e7ff_does_end_time.py

```diff
@@ -0,0 +1,67 @@
+"""does_end_time
+
+Revision ID: 33a17317e7ff
+Revises: caa470b0f83a
+Create Date: 2025-05-10 21:32:56.680280
+
+"""
+
+import sqlalchemy as sa
+from alembic import op
+
+# revision identifiers, used by Alembic.
+revision = "33a17317e7ff"
+down_revision = "caa470b0f83a"
+branch_labels = None
+depends_on = None
+
+
+def upgrade() -> None:
+    # ### commands auto generated by Alembic - please adjust! ###
+
+    # end_date can be derived from existing fields
+    op.add_column(
+        "archive_dynamic_operating_envelope",
+        sa.Column("end_time", sa.DateTime(timezone=True), nullable=False, server_default=sa.text("now()")),
+    )
+    op.execute(
+        "UPDATE archive_dynamic_operating_envelope SET end_time = date_add(start_time, duration_seconds * interval '1 sec', 'UTC')"
```

Comment:
What version of Postgresql are you running? I think this was introduced in 16, but the readme references 14+ being required.

Reply:
The `date_add` function? I'm trying to go over old postgres release notes / feature matrices but I'm struggling to find it. That being said - it's definitely not in postgres 14 :(

Reply:
Released in 16.0

https://www.postgresql.org/docs/release/16.0/

Reply:
I think I'll just have to abandon the generated column and force the application layer to populate it. I'll push up a fix

Reply:
Fix put through in the latest commit